vividus-analytics/src/main/java/org/vividus/analytics/AnalyticsSteps.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.analytics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.eventbus.EventBus;
+
+import org.jbehave.core.annotations.AfterStories;
+import org.jbehave.core.annotations.BeforeStories;
+import org.vividus.analytics.model.AnalyticsEvent;
+import org.vividus.analytics.model.AnalyticsEventBatch;
+import org.vividus.analytics.model.CustomDefinitions;
+import org.vividus.reporter.environment.EnvironmentConfigurer;
+import org.vividus.reporter.environment.PropertyCategory;
+import org.vividus.results.StatisticsProvider;
+import org.vividus.results.model.NodeType;
+import org.vividus.results.model.Statistic;
+
+public class AnalyticsSteps
+{
+    private static final String SESSION_CONTROL = "sc";
+
+    private Stopwatch stopwatch;
+
+    private final EventBus eventBus;
+    private final StatisticsProvider statisticsProvider;
+
+    public AnalyticsSteps(EventBus eventBus, StatisticsProvider statisticsProvider)
+    {
+        this.eventBus = eventBus;
+        this.statisticsProvider = statisticsProvider;
+    }
+
+    @BeforeStories
+    public void reportBeforeStoriesAnalytics()
+    {
+        Map<String, String> payload = new HashMap<>();
+        Map<String, String> configuration = getEnvironmentProperties(PropertyCategory.CONFIGURATION);
+        Map<String, String> modules = getEnvironmentProperties(PropertyCategory.VIVIDUS);
+
+        CustomDefinitions.PROFILES.add(payload, configuration.get("Profiles"));
+        CustomDefinitions.JAVA.add(payload, Runtime.version().toString());
+        CustomDefinitions.VIVIDUS.add(payload, modules.getOrDefault("vividus", "not detected"));
+        CustomDefinitions.REMOTE.add(payload,
+                getEnvironmentProperties(PropertyCategory.PROFILE).get("Remote Execution"));
+        payload.put(SESSION_CONTROL, "start");
+
+        List<AnalyticsEvent> events = new ArrayList<>();
+        events.add(new AnalyticsEvent("startTests", payload));
+        events.addAll(collectPluginsAnalytic(modules));
+
+        eventBus.post(new AnalyticsEventBatch(events));
+
+        stopwatch = Stopwatch.createStarted();
+    }
+
+    @AfterStories
+    public void reportAfterStoriesAnalytics()
+    {
+        long duration = stopwatch.elapsed().toSeconds();
+        Map<String, String> payload = new HashMap<>();
+        Map<NodeType, Statistic> statistic = statisticsProvider.getStatistics();
+        long storiesTotal = statistic.get(NodeType.STORY).getTotal()
+                + statistic.get(NodeType.GIVEN_STORY).getTotal();
```

Comment:
needs clarification

Reply:
in analytics we gathered both given and non-given stories

Reply:
addressed in separate PR